x/gov/client/utils/query.go

```diff
@@ -248,3 +259,36 @@ func QueryProposalByID(proposalID uint64, clientCtx client.Context, queryRoute s
 
 	return res, err
 }
+
+func queryTxsByEvents(clientCtx client.Context, msgType string, otherEvents ...string) (*sdk.SearchTxsResult, error) {
+	events := append([]string{
+		fmt.Sprintf("%s.%s='%s'", sdk.EventTypeMessage, sdk.AttributeKeyAction, msgType),
+	}, otherEvents...)
+
+	// NOTE: SearchTxs is used to facilitate the txs query which does not currently
+	// support configurable pagination.
+	return authclient.QueryTxsByEvents(clientCtx, events, defaultPage, defaultLimit, "")
+}
+
+// searchEvents queries txs by events with both `oldMsgType` and `newMsgtype`,
+// merges the results into one *sdk.SearchTxsResult.
+func searchEvents(clientCtx client.Context, oldMsgType, newMsgType string, otherEvents ...string) (*sdk.SearchTxsResult, error) {
+	// Tx are indexed in tendermint via their Msgs `Type()`, which can be:
+	// - via legacy Msgs (amino or proto), their `Type()` is a custom string,
+	// - via ADR-031 service msgs, their `Type()` is the protobuf FQ method name.
+	// In searching for events, we search for both `Type()`s.
```

Comment:
This is a bugfix that got uncovered when converting CLI to service Msgs.

**Before:**

We only had amino Msgs, their `Type()` was e.g. `"submit_proposal"`. So when searching txs, we were searching `message.action='submit_proposal'`

**Now (since 0.40):**

We have amino Msgs, proto Msgs (`Type()` is same as amino), and service Msgs (`Type()` is proto FQ method name). So for searching txs we need to search for both old and new `Type()`s, e.g.
I'm doing 2 separate searches:
```
message.action='submit_proposal'
// then
message.action='/cosmos.gov.v1beta1.Msg/SubmitProposal'
```

and merging the tx responses.